src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -141,7 +140,17 @@ public ReadOnlyTensorSpan(ReadOnlySpan<T> span, scoped ReadOnlySpan<nint> length
         /// have a rank of 1 and a length equal to the length of the provided <see cref="Array"/>.
         /// </summary>
         /// <param name="array">The target array.</param>
-        public ReadOnlyTensorSpan(Array? array) : this(array, ReadOnlySpan<int>.Empty, array == null ? [0] : (from dim in Enumerable.Range(0, array.Rank) select (nint)array.GetLength(dim)).ToArray(), []) { }
+        public ReadOnlyTensorSpan(Array? array) :
+            this(array,
+                 ReadOnlySpan<int>.Empty,
+                 array == null ?
+                    [0] :
+                    TensorSpanHelpers.FillLengths(array.Rank <= TensorShape.MaxInlineRank ?
+                        stackalloc nint[array.Rank] :
```

Comment:
nit: stack-allocating dynamic lengths has poor performance due to additional checks and guards required. It would be much better to do `stackalloc nint[MaxInlineRank].Slice(array.Rank)`

Reply:
Same general comment applies to other stackallocs as well.

Reply:
> nit: stack-allocating dynamic lengths has poor performance due to additional checks and guards required. It would be much better to do stackalloc nint[MaxInlineRank].Slice(array.Rank)

There are dozens of such existing stackallocs across the library. If we want to replace them with that pattern (with `Slice(0, array.Rank)`, let's do it separately.